File: Calculator1001690798.java
import java.util.Scanner;
import java.lang.System;

public class Calculator1001690798{
	public static void main(String[] args){
		int selection = 0;
		float num1, num2;
		Scanner input = new Scanner(System.in);
		//Get the users calculation choice and the two numbers they will use on the calculation
		while (selection != 5){
			System.out.println("Please pick the operation you would like to perform on the calculator fromm the selection below");
			System.out.println("1: addition");
			System.out.println("2: subtraction");
			System.out.println("3: multiplication");
			System.out.println("4: division");
			System.out.println("5: exit calculator");
			//if the user inputs a bad valid like a string simply tell them to pick a new set of numbers and go back to the start of the selection process
			try{
				selection = input.nextInt();
				if(selection == 5){
					System.exit(0);
				}
				System.out.println("Please pick two numbers you would like to perform this on");
				num1 = input.nextFloat();
				num2 = input.nextFloat();
			}
			catch(Exception e){
				System.out.println("\nPlease enter only numbers\n");
				input.nextLine();
				continue;
			}
			switch(selection){
				case 1:
					addition(num1, num2);
					break;
				case 2:
					subtraction(num1, num2);
					break;
				case 3:
					multiplication(num1, num2);
					break;
				case 4:
					division(num1, num2);
					break;
			}
		}
	}
	//Encapsulated functions for performing the selected calculation
	private static void addition(float a, float b){
		System.out.printf("%.1f \n", a + b);
	}

	private static void subtraction(float a, float b){
		System.out.printf("%.1f \n", a - b);
	}

	private static void multiplication(float a, float b){
		System.out.printf("%.1f \n", a * b);
	}

	private static void division(float a, float b){
		if(b == 0){
			System.out.println("Please don't divide by zero\n");
			return;
		}	
			System.out.printf("%.1f \n", a / b);
	}
}

File: Calculator1001690798.py
import sys

VALID_OPERATIONS = ["+","-","*","/"]
def main():
    while True:
        #Get the users choice of calculation
        print("Please pick the operation you would like to perform(+ - * /) or q to quit");
        try:
            selection = str(input())
            if(selection == "q"):
                sys.exit()
            #Check that the user has selected a valid operation from the list
            elif(any(selection in operation for operation in VALID_OPERATIONS) == False):
                print("\nPlease pick a valid operation\n")
                continue
            #Get the values of the numbers the user would like to perform the calculation on
            print("Please enter two numbers you would like to perform this operation on")
            num1 = float(input())
            num2 = float(input())
        #Raise an exception if the user trys to enter a value that isnt a number like a string
        except Exception as e:
            print("Please enter only numbers\n")
            continue
        if (selection == "+"):
            addition(num1, num2)
        elif (selection == "-"):
            subtraction(num1, num2)
        elif (selection == "*"):
            multiplication(num1, num2)
        elif (selection == "/"):
            division(num1, num2)

#Functions to perform the users desired calculation
def addition(num1, num2):
    print("%.1f\n" %(num1 + num2))

def subtraction(num1, num2):
    print("%.1f\n" %(num1 - num2))

def multiplication(num1, num2):
    print("%.1f\n" %(num1 * num2))

def division(num1, num2):
    #Check that the user isnt trying to divide by zero
    if(num2 == 0):
        print("Please dont divide by zero\n")
        return
    print("%.1f\n" %(num1 / num2))

if __name__ == '__main__':
    main()
